[Toh Kar Wi] iP

CrownKira opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

CrownKira committed:
Ignore .class files

CrownKira committed:
Implement Level-1

CrownKira committed:
Implement Level-2

CrownKira committed:
Implement Level-3 and A-Classes

CrownKira committed:
Implement Level 4

CrownKira committed:
Implement Level-4

CrownKira committed:
Implement Level-5

CrownKira committed:
Implement Level-5

CrownKira committed:
Update EXPECTED.TXT

CrownKira committed:
Implement Level-6

CrownKira committed:
Add todo

CrownKira committed:
Implement A-Enums

CrownKira committed:
Format code

CrownKira committed:
Implement Level 7

CrownKira committed:
Implement Level 7

CrownKira committed:
Implement Level 8

CrownKira committed:
Merge branch 'branch-Level-7'

CrownKira committed:
Resolve merge conflicts

CrownKira committed:
Refactor exceptions

CrownKira committed:
Implement A-MoreOOP

CrownKira committed:
Implement A-Packages

CrownKira committed:
Create test folders

CrownKira committed:
Scaffold tests

CrownKira committed:
Add unittest for parser

CrownKira committed:
Implement A-Jar

CrownKira committed:
Implement A-JavaDoc

CrownKira committed:
Add headers to class

CrownKira committed:
Implement A-CodingStandard

CrownKira committed:
Merge branch 'A-CodingStandard'

CrownKira committed:
Merge branch 'branch-A-JavaDoc'

CrownKira committed:
Implement Level-9

CrownKira committed:
Merge branch 'branch-Level-9'

nujiak commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,31 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task task = new Todo(description);
+        tasks.add(task, storage);
+        System.out.println("Got it. I've added this task:\n  " +
+                task +
+                "\nNow you have " + tasks.toArray().length + " task(s) in the list.");
```
Should the line break come before the + operator instead?

CrownKira replied:
Oops. Forgot to configure the formatter to do this.

nujiak commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,59 @@
+package duke;
+
+import duke.command.*;
```
Should this wildcard import be replaced by individual classes instead?

CrownKira replied:
Good point. I will change it!

nujiak commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,59 @@
+        Scanner sc = new Scanner(fullCommand);
+        String input = sc.next();
+        try {
+            switch (input) {
+            case "list": {
+                return new ListCommand();
+            }
+            case "done": {
+                return new DoneCommand(sc.nextInt());
+            }
+            case "remove": {
+                return new DeleteCommand(sc.nextInt());
+            }
+            case "todo": {
+                return new TodoCommand(sc.nextLine().trim());
+            }
+            case "deadline": {
+                return new DeadlineCommand(sc.nextLine().trim());
+            }
+            case "event": {
+                return new EventCommand(sc.nextLine().trim());
+            }
+            case "find": {
+                return new FindCommand(sc.nextLine().trim());
+            }
+            case "bye": {
+                return new ExitCommand();
+            }
+            default: {
+                throw new DukeParseException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+            }
```
I like how you used switch-case to keep your code tidy and readable

hpkoh commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,26 @@
+     * Constructs a DukeException with no detailed message.
+     */
+    public DukeException() {
+        super();
```
Not really an issue but I believe super() will be called by default if no superclass constructor is called!

CrownKira replied:
Ahh nice point! I will change it  :))

hpkoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+    /**
+     * Deadline
+     */
+    protected LocalDate by;
```
Perhaps the name of this field can be more descriptive, such as ```deadlineDate``` ? The comment however does clearly give enough information about the field.

CrownKira replied:
True. I will make it more explicit!

hpkoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+        } catch (DateTimeParseException e) {
+            throw new DukeIllegalFormatException(
+                    "☹ OOPS!!! Seems like you have entered a wrong date format. " +
+                            "Try this instead: YYYY-MM-DD"
```
Should the indentation of this line be the same as line 39 as well? I believe it is ok for all wrapped lines to be at the same indentation level!

CrownKira replied:
Hmm, I have been letting the formatter do the work haha. Will take note of it next time!

hpkoh commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,17 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Perhaps explicit imports can be used instead of the wildcard imports?

CrownKira replied:
Good catch. Thanks and I will fix it !!

hpkoh reviewed (commented):
Your code is very well-formatted with descriptive and relevant javadocs! The readability is impressive and I could only really find some minor nitpicks to comment on.

CrownKira committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2122S1/ip into add-gradle-support

CrownKira committed:
Merge branch 'add-gradle-support'

CrownKira committed:
Add configuration files for checkstyle

CrownKira committed:
Modify mainClassName in build.gradle

CrownKira committed:
Reformat code

CrownKira committed:
Fix OperatorWrap checkstyle

CrownKira committed:
Reformat code to pass checkstyle tests

CrownKira committed:
Implement GUI

CrownKira committed:
Implement basic GUI

CrownKira committed:
Improve GUI to allow interaction with Duke

CrownKira committed:
Merge branch 'branch-GUI'

CrownKira committed:
Refactor duke gui and ui logic

CrownKira committed:
Implement A-Assertions

CrownKira committed:
Merge pull request #1 from CrownKira/branch-A-Assertions

Branch A Assertions

CrownKira committed:
Implement A-CodeQuality

CrownKira committed:
Merge branch 'master' into branch-A-CodeQuality

CrownKira committed:
Merge pull request #2 from CrownKira/branch-A-CodeQuality

Branch A-CodeQuality

CrownKira committed:
Implement A-Streams

CrownKira committed:
Merge pull request #3 from CrownKira/A-Streams

Branch A-Streams

CrownKira committed:
Create gradle.yml

CrownKira committed:
Reset README.md

CrownKira committed:
Modify bye message

CrownKira committed:
Fix checkstyle

CrownKira committed:
Fix validateRow

CrownKira committed:
Implement BCD-Extension

CrownKira committed:
Add tagging feature

Tagging system is absent.

Tags make filtering, finding and grouping of relevant tasks more efficient.

Here is a list of sub features introduced:
- add tags with todo, event and deadline command using /tag flag
- save tags to db
- load tags from db
- show tags in list items

Other features such as: update, remove and find tags will be introduced in future releases.

CrownKira committed:
Fix checkstyle

CrownKira committed:
Fix list all

CrownKira committed:
Implement A-BetterGui

CrownKira committed:
Rename product to ShyBot

CrownKira committed:
Modify data

CrownKira committed:
Add Ui.png to docs

CrownKira committed:
Remove unused files

CrownKira committed:
Fix jar launcher issue

CrownKira committed:
Rectify format issue

CrownKira committed:
Implement A-UserGuide

CrownKira committed:
Fix checkstyle error

CrownKira committed:
Remove unused image

CrownKira committed:
Add image to README.md

CrownKira committed:
Make minor correction to README.md
